src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -120,6 +125,58 @@ private void WriteClientMethodBody(CodeWriter writer, LowLevelClientMethod clien
             writer.Line();
         }
 
+        private string BuildSchemaFromDocs(LowLevelClientMethod.SchemaDocumentation[] docs, bool showRequired)
+        {
+            var docDict = docs.ToDictionary(d => d.SchemaName, d => d);
+            var builder = new StringBuilder();
+            builder.AppendLine("{");
+            BuildSchemaFromDoc(builder, docs.FirstOrDefault(), docDict, showRequired, 2);
+            builder.AppendLine("}");
+            return builder.ToString();
+        }
+
+        private void BuildSchemaFromDoc(StringBuilder builder, LowLevelClientMethod.SchemaDocumentation doc, IDictionary<string, LowLevelClientMethod.SchemaDocumentation> docDict, bool showRequired, int indentation = 0)
+        {
+            foreach (var row in doc.DocumentationRows)
+            {
+                var required = showRequired && row.Required ? " (required)" : string.Empty;
+                var isArray = row.Type.EndsWith("[]");
+                var rowType = isArray ? row.Type.Substring(0, row.Type.Length - 2) : row.Type;
+                builder.AppendIndentation(indentation).Append($"{row.Name}: ");
+                if (isArray)
+                {
+                    if (docDict.ContainsKey(rowType))
+                    {
+                        builder.AppendLine("[");
+                        var docToProcess = docDict[rowType];
+                        docDict.Remove(rowType); // In the case of cyclic reference where A has a property type of A itself, we just show the type A if it's not the first time we meet A.
+                        builder.AppendIndentation(indentation + 2).AppendLine("{");
+                        BuildSchemaFromDoc(builder, docToProcess, docDict, showRequired, indentation + 4);
+                        builder.AppendIndentation(indentation + 2).AppendLine("}");
+                        builder.AppendIndentation(indentation).AppendLine($"]{required},");
+                    }
+                    else
+                        builder.AppendLine($"[{rowType}]{required},");
+                }
+                else
+                {
+                    if (docDict.ContainsKey(rowType))
+                    {
+                        builder.AppendLine("{");
+                        var docToProcess = docDict[rowType];
+                        docDict.Remove(rowType); // In the case of cyclic reference where A has a property type of A itself, we just show the type A if it's not the first time we meet A.
+                        BuildSchemaFromDoc(builder, docToProcess, docDict, showRequired, indentation + 2);
+                        builder.AppendIndentation(indentation).Append("}").AppendLine($"{required},");
+                    }
+                    else
+                        builder.AppendLine($"{rowType}{required},");
+                }
+            }
+            // Remove the last "," by first removing ",\n", then add back "\n".
+            builder.Length -= 3;
```

Comment:
Suspect this should be `bulder.Length -= 1 + Environment.NewLine.Length` (since the length of the appended newline will differ between Unix and Windows)

Reply:
`writer.RemoveTrailingComma()`

Reply:
Good catch. You saved my time to reproduce the error in Linux!